xla/hlo/ir/backend_config.cc

```diff
@@ -57,6 +60,13 @@ const std::string& BackendConfigWrapper::GetRawStringWithoutMutex() const {
   return raw_string_.empty() ? *kEmptyString : raw_string_;
 }
 
+absl::StatusOr<std::string> BackendConfigWrapper::GetFingerprint() const {
+  absl::MutexLock lock{&mutex_};
+  std::string result;
+  TF_RET_CHECK(tsl::SerializeToStringDeterministic(*proto_, &result));
```

Comment:
This may be `null`. The backend config is initialized with either one of `proto` or `raw_string` and the other one is not set unless you call a getter method for it.

To fix:
- Remove the locking of the mutex above and directly call the `GetProto()` method.
- Add a test that also tests the path where the backend config is created from a string, not a proto.

Reply:
How does one call GetProto(&proto) for an unknown proto? (tsl::protobuf::Message is abstract)

Reply:
For our purpose fingerprinting of unknown protos created from strings is not required, I can just add a proto_ != nullptr check and a test if that's OK.

Reply:
This will be brittle. If we make changes to `BackendConfigWrapper` they should work in all cases. See my other comment.